### PR TITLE
Adding ability to sell a piece of race merchandise when entering

### PR DESCRIPTION
**Pull request by billwright**
@moveson can you take a look at this?

My tests are failing, but I don't know why. It doesn't seem related to my change, but, of course, it must be... Except that the tests pass on GitHub! What's happening here?

Here's the error (this is running 'rake test') (see below).

Ah, I think the error must be related to adding fields to the race_edition model and that must have broken the show view for it. But the error about missing :id is confusing... I didn't change anything with id...

Also, the site starts and operates without errors in my manual checks...so far.

(base) rattlesnake_ramble> test 
Running via Spring preloader in process 40496
Running via Spring preloader in process 40499
Run options: --seed 24833

Running:

E

Error:
WelcomeControllerTest#test_should_get_index:
ActionView::Template::Error: No route matches {:action=>"show", :controller=>"race_editions", :id=>nil}, missing required keys: [:id]
    app/views/layouts/_navigation.html.erb:14
    app/views/layouts/application.html.erb:13
    test/controllers/welcome_controller_test.rb:5:in `block in <class:WelcomeControllerTest>'

rails test test/controllers/welcome_controller_test.rb:4

### CODE REVIEW

File: db/migrate/20240625234938_add_merchandise_to_race_editions.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class AddMerchandiseToRaceEditions < ActiveRecord::Migration[7.0]
+  def change
+    add_column :race_editions, :selling_merchandise, :boolean, default: false
+    add_column :race_editions, :merchandise_description, :string, null: true
+    add_column :race_editions, :merchandise_image_file_name, :string, null: true
```
Comment (moveson) [inline db/migrate/20240625234938_add_merchandise_to_race_editions.rb:5]: Do we want to use the Rails built-in attachments feature for images? It's pretty nice.

File: test/controllers/welcome_controller_test.rb [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 class WelcomeControllerTest < ActionDispatch::IntegrationTest
   test "should get index" do
+    # TODO: what is this welcome_index_url? This text only appears in this one spot in the project
```
Comment (moveson) [inline test/controllers/welcome_controller_test.rb:5]: Yep, `welcome_index_url` is a Rails-generated URL for `GET welcome/index`

You can see all of the url helpers by going to a terminal prompt and entering `rails routes`. (Use a very wide terminal window to see the output clearly.)

**Review by moveson — APPROVED** [review]
We should consider using Rails' built-in attachments for the merch images. Otherwise looks good to me.

File: test/controllers/welcome_controller_test.rb (hunk 2 above)
Comment (billwright) [reply test/controllers/welcome_controller_test.rb:5]: Thanks

File: db/migrate/20240625234938_add_merchandise_to_race_editions.rb (hunk 1 above)
Comment (billwright) [reply db/migrate/20240625234938_add_merchandise_to_race_editions.rb:5]: Sure. If there is a better/more Rails way to do it, let's do it! Can you point me to some info on this approach?

Comment (billwright) [reply db/migrate/20240625234938_add_merchandise_to_race_editions.rb:5]: Actually, let's wait on that. It sounds like a cleaner way to go -- creating a new ActiveRecord for merchandise, but without a separate record for the merchandise doesn't seem like using the attachments to the RaceEdition is best... but what do I know? I just looked for about 5 minutes. :-) We can always clean this up later.

**Pull request merged**